web/modules/custom/dpl_opening_hours/dpl_opening_hours.module

```diff
@@ -47,6 +52,79 @@ void {
   }
 }
 
+/**
+ * Implements hook_form_FORM_ID_alter() for the taxonomy term deletion form related to opening hours.
+ *
+ * This hook modifies the deletion confirmation form for taxonomy terms within
+ * the "opening_hours_categories" vocabulary, adding custom messages.
+ *
+ * @param array $form
+ *   The form structure array.
+ * @param \Drupal\Core\Form\FormStateInterface $form_state
+ *   The current state of the form.
+ * @param string $form_id
+ *   The ID of the form being altered.
+ */
+function dpl_opening_hours_form_taxonomy_term_opening_hours_categories_delete_form_alter(array &$form, FormStateInterface $form_state, string $form_id): void {
+  $form_object = $form_state->getFormObject();
+  if (!$form_object instanceof EntityFormInterface) {
+    return;
+  }
+
+  $term = $form_object->getEntity();
+  if (!$term instanceof Term) {
```

Comment:
```suggestion
  if (!$term instanceof TermInterface) {
```

When we want to assert something from a value [we generally try to use the interfaces instead of the class](https://www.drupal.org/docs/develop/standards/php/object-oriented-code#interfaces) when an appropriate interface exists.

Remember to update your imports accordingly.